src/net/download.rs

```diff
@@ -149,10 +157,40 @@ impl BlockRequestKey {
             child_block_header: child_block_header,
             sortition_height: sortition_height,
             download_start: get_epoch_time_secs(),
+            kind,
+        }
+    }
+}
+
+impl Requestable for BlockRequestKey {
+    fn get_url(&self) -> &UrlString {
+        &self.data_url
+    }
+
+    fn get_request_type(&self, peer_host: PeerHost) -> HttpRequestType {
+        match self.kind {
+            BlockRequestKeyKind::Block => HttpRequestType::GetBlock(
+                HttpRequestMetadata::from_host(peer_host),
+                self.index_block_hash,
+            ),
+            BlockRequestKeyKind::ConfirmedMicroblock => HttpRequestType::GetMicroblocksConfirmed(
+                HttpRequestMetadata::from_host(peer_host),
+                self.index_block_hash,
+            ),
         }
     }
 }
 
+impl std::fmt::Display for BlockRequestKey {
```

Comment:
Please print out the same contents as `std::fmt::Debug` (or remove this `impl` outright).

Reply:
The amount of informations displayed here is exactly what you were already displaying (line 1734 - same file). Do you need more?